peer-verification/src/signer.rs

```diff
@@ -0,0 +1,52 @@
+use clap::Parser;
+use libp2p::identity::{ed25519 as libp2p_ed25519, PublicKey};
+use std::{fs, path::Path};
+
+#[derive(Debug, Parser)]
+#[clap(version = "1.0")]
+pub struct Config {
+    /// Message we want to sign.
+    #[clap(long)]
+    pub message: String,
+
+    /// Path to p2p secret.
+    #[clap(long)]
+    pub p2p_secret_path: String,
+}
+
+fn derive_key(bytes: Vec<u8>) -> Option<libp2p_ed25519::SecretKey> {
+    libp2p_ed25519::SecretKey::from_bytes(&mut bytes.clone())
+        .ok()
+        .or_else(|| {
+            let mut decoded = hex::decode(bytes).ok()?;
+            libp2p_ed25519::SecretKey::from_bytes(&mut decoded).ok()
+        })
+}
+
+fn main() {
+    let Config {
+        message,
+        p2p_secret_path,
+    } = Config::parse();
+
+    let path = Path::new(&p2p_secret_path);
+    if !path.exists() {
+        panic!("Can not find p2p secret file: {:?}", p2p_secret_path);
```

Comment:
I see you're using past tense for errors throughout the file, so perhaps it would be best to stick to this convention (i.e. `Could not find [...]`)